src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -820,14 +837,9 @@ bool TreeCtrl::Remove(const CUUID &uuid)
     // group as well. repeat up the tree...
     wxTreeItemId parentId = GetItemParent(id);
     Delete(id);
-    while (parentId != GetRootItem()) {
-      wxTreeItemId grandparentId = GetItemParent(parentId);
-      if (GetChildrenCount(parentId) == 0) {
-        Delete(parentId);
-        parentId = grandparentId;
-      } else
-        break;
-    } // while
```

Comment:
What happened to the tree traversing loop? Is it no longer needed?

Reply:
The tree traversing loop is no longer needed. It is faulty in some cases and therefore had been corrected (removed). In case an element is removed, e.g.
+ -
|_ A group
      |_ B group
            |_ C entry
The recursive deletion would remove group A and B from the tree view in case C entry is deleted. Internally the group A and B are still present and will come back when tree view is rebuild (e.g. after coming back from grid view).